app/halgo/Halgo/CLA/Command/Indexer.hs

```diff
@@ -0,0 +1,51 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+-- | Communicate with indexer commands.
+module Halgo.CLA.Command.Indexer
+  ( indexerOpts
+  ) where
+
+
+import Control.Monad.Reader (asks)
+import Fmt ((+|), (|+))
+import Options.Applicative (Parser, command, help, hsubparser, info, long, metavar, optional,
+                            progDesc, short, strOption)
+
+import Network.Algorand.Client (Host)
+
+import Halgo.CLA.Type (MonadSubCommand, SubCommand, goNetwork)
+import Halgo.IO (putTextLn)
+import Halgo.Util (die)
+
+optIndexerHost :: Parser (Maybe Host)
+optIndexerHost = optional . strOption $ mconcat
+  [ long "indexer-host"
+  , short 'i'
+  , metavar "INDEXER_HOST"
+  , help "HOST of the indexer to connect to (default: AlgoExplorer node based on the chosen network)"
+  ]
+
+indexerOpts :: Parser SubCommand
+indexerOpts = cmdIndexer <$> optIndexerHost <*> hsubparser (mconcat
+  [ command "host"
+    $ info (pure cmdIndexerHost)
+    $ progDesc "Show the HOST of the indexer that will be used"
+
+  ])
+
+cmdIndexer :: MonadSubCommand m => Maybe Host -> (Host -> m ()) -> m ()
+cmdIndexer url sub = getIndexerHost url >>= sub
+  where
+    getIndexerHost :: MonadSubCommand m => Maybe Host -> m Host
+    getIndexerHost (Just u) = pure u
+    getIndexerHost Nothing = asks goNetwork >>= \case
+      "mainnet-v1.0" -> pure "http://indexer.algoexplorerapi.io/"
+      "testnet-v1.0" -> pure "http://indexer.testnet.algoexplorerapi.io/"
+      "betanet-v1.0" -> pure "http://indexer.betanet.algoexplorerapi.io/"
```

Comment:
1. Shouldn't it be https?
2. I think ideally we should define a common function (or a map) to be used here and in `getNodeHost`, so that network names appear in only one place and it's easier to match node and indexer URLs. Currently we can add a new network into one place and forget about the other one.

Reply:
1. Yes!
2. Good idea